doc/mgmt/gnmi/SONiC_GNMI_Server_Interface_Design.md

```diff
@@ -441,18 +441,42 @@ For incremental configuration, 'config apply-patch' should not restart gNMI, bgp
 
 ##### 1.2.1.11.2 Full Configurations
 
-For full configuration, there’re 2 possible solutions: 
-* gNMI server needs to send response at first, and then invoke 'config reload'.
-
-<img src="images/full_rpc.svg" alt="full-gnmi" width="800px"/>
-
-* Use gNMI request and gNOI request together to implement full configuration update.
-
-<img src="images/full_rpc_gnoi.svg" alt="full-gnmi-gnoi" width="800px"/>
-
-The full configuration request will be overwritten by subsequent full configuration request or incremental configuration request.
-
-<img src="images/mixed requests.svg" alt="overwritten-config" width="800px"/>
+We will use single gnmi request to support full configuration update.
+1. The host service will perform YANG validation and return an error if it fails.
```

Comment:
In cases where we need to enforce custom validation rules that go beyond the constraints of the Yang model, how does the current design plan to handle these? translib based infrastructure provides this mechanism.

Reply:
You can continue using the current translib infrastructure, and this change will not impact it. We are utilizing the origin field of the gnmi request, and only those requests with the 'sonic-db' origin field can use this gnmi full config update.

Reply:
I agree, we don't have a good solution at the moment. Let's keep this limitation, as gNMI can't be used to change the management network or gNMI server settings.